load.c

```diff
@@ -597,7 +597,7 @@ rb_feature_p(rb_vm_t *vm, const char *feature, const char *ext, int rb, int expa
 
     loading_tbl = get_loading_table(vm);
     f = 0;
-    if (!expanded) {
+    if (!expanded && !rb_is_absolute_path(feature)) {
```

Comment:
What occurs when `feature` is an absolute path but not a real path, i.e., symlink?

Reply:
I don't think it change anything? At least on profiles I can't see Ruby calling `realpath` even before that change, but maybe I missed it?

Reply:
According to tests I did a while back:

* `Kernel#require` seems to base idempotence on real paths.
* `$LOADED_FEATURES` shows what the user entered.
  * If it is an absolute path, it remains as it is verbatim.
  * If it was `require "foo"`, the `foo` itself remains verbatim (mod extension), and the load path directory that acts as prefix appears as its real path.

I don't know if that logic applies here too, sharing in case it does.

Reply:
My words were not enough.
When a library has been loaded by `$LOAD_PATH`, requiring a symlink to that path will do nothing now.
I think this PR will change it to load the same file again, no?

Reply:
@nobu sorry but I'm really not sure what you have in mind, perhaps you could provide a test script?

I tried a few things:

```ruby
Dir.mkdir("/tmp/original") rescue nil
File.write("/tmp/original/a.rb", "p :loaded_original_a")

File.symlink("/tmp/original", "/tmp/symlink") rescue nil
$LOAD_PATH << "/tmp/original"

require "a"

$LOAD_PATH << "/tmp/original/symlink"

require "a"
```

```ruby
File.write("/tmp/a.rb", "p :loaded_a")
File.symlink("/tmp/a.rb", "/tmp/b.rb")
$LOAD_PATH << "/tmp"

require "a"
require "b"
```

None changed.

Reply:
The case `require "a"` then `require "/tmp/original/symlink/a"`.
In the latter `require`, the path will be expanded to the original path, and not be loaded again, for now.
This PR skips that expansion, and the file will be loaded again, no?

Reply:
I just tried:

```ruby
Dir.mkdir("/tmp/original") rescue nil
File.write("/tmp/original/a.rb", "p :loaded_original_a")

File.symlink("/tmp/original", "/tmp/symlink") rescue nil
$LOAD_PATH << "/tmp/original"

require "a"

$LOAD_PATH << "/tmp/symlink"

require "/tmp/symlink/a"
```

And the file is still only loaded once.